pinot-core/src/main/java/com/linkedin/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -203,6 +203,13 @@ public BrokerResponseNative reduceOnDataTable(@Nonnull BrokerRequest brokerReque
           setGroupByHavingResults(brokerResponseNative, aggregationFunctions, aggregationFunctionSelectStatus,
               brokerRequest.getGroupBy(), dataTableMap, brokerRequest.getHavingFilterQuery(),
               brokerRequest.getHavingFilterSubQueryMap());
+          if (brokerMetrics != null) {
+            long groupBySize = 0L;
+            for (AggregationResult aggregationResult : brokerResponseNative.getAggregationResults()) {
+              groupBySize = Math.max(aggregationResult.getGroupByResult().size(), groupBySize);
```

Comment:
Can you explain why using the maximum group-by size from servers as the metric value?